File: app/discover/db_fetch_az_hosts.rb
require_relative 'db_access'

class DbFetchAZHosts < DbAccess
  
  def get(id)
    query = %Q{
      SELECT DISTINCT host, host AS id
      FROM nova.instances
      WHERE availability_zone = '#{id}'
        AND host IS NOT NULL
        AND availability_zone IS NOT NULL
        AND deleted = 0
    }
    return get_objects_list(query, "host")
  end
  
end

File: app/discover/db_fetch_az_hosts.py
from db_access import DbAccess

class DbFetchAZHosts(DbAccess):
  
  def get(self, id):
    query = """
      SELECT DISTINCT host, host AS id, host_ip AS ip_address
      FROM nova.instances i
        JOIN nova.compute_nodes n ON i.node = n.hypervisor_hostname
      WHERE availability_zone = %s
        AND host IS NOT NULL
        AND i.deleted = 0
    """
    results = self.get_objects_list_for_id(query, "host", id)
    for r in results:
      r["id"] = r["host"]
      r["host_type"] = "Compute node"
    return results
